test/end2end_test.go

```diff
@@ -7819,3 +7819,36 @@ func (s) TestStreamingServerInterceptorGetsConnection(t *testing.T) {
 		t.Fatalf("ss.Client.StreamingInputCall(_) = _, %v, want _, %v", err, io.EOF)
 	}
 }
+
+func unaryInterceptorVerifyPost(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Error(codes.NotFound, "metadata was not in context")
+	}
+	method := md.Get(":method")
+	if len(method) != 1 {
+		return nil, status.Error(codes.NotFound, ":method value had more than one value")
+	}
+	if method[0] != "POST" {
+		return nil, status.Error(codes.NotFound, ":method value was not post")
+	}
+	return nil, status.Error(codes.OK, "")
+}
+
+// TestUnaryInterceptorGetsPost verifies that the server transport adds a
+// :method POST header to metadata, and that that added Header is visibile at
+// the grpc layer.
+func (s) TestUnaryInterceptorGetsPost(t *testing.T) {
+	ss := &stubserver.StubServer{}
+	if err := ss.Start([]grpc.ServerOption{grpc.UnaryInterceptor(unaryInterceptorVerifyPost)}); err != nil {
```

Comment:
You don't need an interceptor. Just read and check the metadata in `StubServer.EmptyCall`.

Reply:
Or, actually, it is OK. The purpose is to test interceptors.

Reply:
Yeah, because it's in the context of RBAC. Agree with you otherwise though.